core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -470,6 +470,19 @@ public WearingItemFilter parseWearingItem(Element el) throws InvalidXMLException
     return new WearingItemFilter(factory.getKits().parseRequiredItem(el));
   }
 
+  @MethodParser("effect")
+  public HasEffectFilter parseHasEffect(Element el) throws InvalidXMLException {
+    Attribute minAttribute = el.getAttribute("min-duration");
```

Comment:
Make use of PGM nodes  instead of XML attributes, eg:: `XMLUtils.parseDuration(Node.fromAttr(el, "min-duration"))`